src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/Http2Connection.cs

```diff
@@ -844,25 +844,18 @@ private void ProcessGoAwayFrame(FrameHeader frameHeader)
 
         private abstract class WriteQueueEntry : TaskCompletionSource
         {
-            private readonly CancellationToken _cancellationToken;
             private readonly CancellationTokenRegistration _cancellationRegistration;
 
             public WriteQueueEntry(int writeBytes, CancellationToken cancellationToken)
                 : base(TaskCreationOptions.RunContinuationsAsynchronously)
             {
                 WriteBytes = writeBytes;
 
-                _cancellationToken = cancellationToken;
-                _cancellationRegistration = cancellationToken.UnsafeRegister(static s => ((WriteQueueEntry)s!).OnCancellation(), this);
+                _cancellationRegistration = cancellationToken.UnsafeRegister(static (s, cancellationToken) => ((WriteQueueEntry)s!).TrySetCanceled(cancellationToken), this);
```

Comment:
Did you mean to change this from SetCanceled to TrySetCanceled? It was intentionally SetCanceled because it should never fire after we've completed the TCS.

Reply:
I did, since it shouldn't matter, but I can change it back if you'd prefer it actually throw back to the Cancel if something goes wrong.

Reply:
I don't have a strong preference either way. That said, using SetCanceled makes it clear that the TCS should not have been completed previously. If you change it to TrySetCanceled, please assert the result for the same reason.

Reply:
Added assert.